### ISSUE TITLE
buffer moves to top of the file then back to cursor with each save 

### ISSUE THREAD

**Issue by andreiamatuni** [fatih/vim-go#1036]
Every time I save a file (command-S in MacVim or :w) the page moves to the top of the file then moves back to where the cursor was before saving. It didn't used to do this. Maybe something to do with gofmt or goimports? How do I make it so that the buffer stays in place when I save?

vim version:
VIM - Vi IMproved 7.4 (2013 Aug 10, compiled Aug 29 2016 12:51:13)
MacOS X (unix) version
Included patches: 1-2290
Compiled by Homebrew
vim: stable 7.4.2290 (bottled), HEAD

go version: 1.7

vim-go version: master

**Comment by nhooyr** [comment]
That is `:GoFmt` iirc.

**Comment by andreiamatuni** [comment]
It used to be able to format the code without moving the page up and down though. How do I get it to behave like that?

**Comment by nhooyr** [comment]
See #760.

**Comment by andreiamatuni** [comment]
yeah i read that issue already. 

I changed g:go_fmt_command from "goimports" to "gofmt" and now it doesn't do the page movement thing anymore. Looks like it's goimports' fault, or something to do with how vim-go deals with goimports (since this doesn't happen in Atom, using goimports)

**Comment by nhooyr** [comment]
I could be wrong but I think they are handled the same. Changing the command only changes what executable the file is fed to.

**Comment by andreiamatuni** [comment]
actually nevermind it's still doing it...ugh

**Comment by mhf-air** [comment]
I have exactly the same problem, right after I updated from vim7.4 to vim8.0 today. How to fix it?

**Comment by mhf-air** [comment]
well, it's not vim8.0's problem. I just tested it on vim7.4, same thing happens again.

**Comment by fatih** [comment]
The cursor is where it is, only if you use `goimports`, then there is a chance it can move a couple of lines because of adding new imports. Otherwise there is no issue. The issue @nhooyr mentioned is not tackled and thus we currrently only use vimscript based format. Thanks for the issue, closing it.

**Comment by andreiamatuni** [comment]
It's not a couple of lines. This isn't about the buffer being updates because new imports were added at the top. It's about the view of the entire page being moved to the top of the page and then immediately back to where your cursor was when you saved. Is this because of vim 8.0? I don't remember if I had this problem when I was still on 7.4, but I think I did. The other person in this thread mentioned it started happening to them after they updated to 8.0.
